Cookbook/Technical-Documents/CRP-Bot.md

```diff
@@ -0,0 +1,99 @@
+
+One consequence of that process is that – because we use a monorepo both on Android and iOS (i.e. the codebase is shared amongst all the app flavors for iOS, and similar for Android) – **JIRA tickets that are implementing a feature for one flavor (e.g. Telus or Ascension) will (also) be marked with versions for other flavors (e.g. "iOS Babylon 4.4.0")** which also include that commit
+
+This could be surprising at first, but is in fact totally normal and expected:
+
+* This allows us to track that the commit (and thus changes in the code) that implemented that feature ended up being merged in the develop branch before we cut the release for Babylon UK and thus the code would be included in what ends in the UK app in the stores.
+* This is important to track even if the ticket in question is supposed to only affect the US app, because ISO and SSDLC is about tracking – including being able to locate when a regression has been introduced and in which releases it might have been shipped – and it could happen that the **commit implementing that feature for the US app accidentally had unforeseen side-effect on the UK** app (e.g. changing something in a module that is used by both flavors).
+* Even if that side effect is unintentional, we need to know that it ended up shipped in the UK release. (Even if admittedly in the ideal case the ticket should not have affected the UK release, but bugs and unintended side-effects happen)
+
+Note however that even if a US ticket will be flagged with e.g. "iOS Babylon 4.5.0" (if corresponding PR was merged before the UK 4.5.0 release), it will eventually also be flagged with e.g. the "iOS Ascension 1.1.0" version in JIRA too once it's actually shipped as part of the US release.
+
+This means that a ticket will ultimately end up with multiple versions in its "Fix Versions" field (typically for different flavors and platforms), allowing us to know all the released versions in which the code related to that ticket ended up, even if it was dormant/inactive/disabled for some flavors.
+
+Again, this might seem surprising at first, but this is expected and on purpose.
+
+### Impact on "Fix versions" vs feature flags
+
+Another impact of that process is that **even features that are gated by a local feature switch** (hardcoded as being turned off in code) and end up still being disabled when we release the app **will still be flagged with the released version in their "Fix Version"**.
+
+This is because – similar to what was described above with the example of unwanted side effects of a US feature on the UK app – the implementation of a feature that is theoretically supposed to be disabled by a feature flag... can accidentally introduce bugs or have some of the code not protected behind the feature flag like it should have been. So since the code is present in the final app anyway – even if it's supposed to be disabled – then the corresponding JIRA ticket will still have the version set in its "Fix Version".
```

Comment:
```suggestion
This is because – similar to what was described above with the example of unwanted side effects of a US feature on the UK app – the implementation of a feature that is theoretically supposed to be disabled by a feature flag can accidentally introduce bugs or have some of the code not protected behind the feature flag like it should have been. So since the code is present in the final app anyway – even if it's supposed to be disabled – the corresponding JIRA ticket will still have the version set in its "Fix Version".
```